vision/google/cloud/vision/_gax.py

```diff
@@ -39,21 +39,21 @@ def annotate(self, image, features):
         :type features: list
         :param features: List of :class:`~google.cloud.vision.feature.Feature`.
 
-        :rtype: :class:`~google.cloud.vision.annotations.Annotations`
-        :returns: Instance of ``Annotations`` with results or ``None``.
+        :rtype: :class:`~google.cloud.vision.annotations.Annotations` or list
+        :returns: Instance of ``Annotations`` or list of ``Annotations``.
         """
         gapic_features = [_to_gapic_feature(feature) for feature in features]
         gapic_image = _to_gapic_image(image)
         request = image_annotator_pb2.AnnotateImageRequest(
             image=gapic_image, features=gapic_features)
         requests = [request]
         annotator_client = self._annotator_client
-        images = annotator_client.batch_annotate_images(requests)
-        if len(images.responses) == 1:
-            return Annotations.from_pb(images.responses[0])
-        elif len(images.responses) > 1:
-            raise NotImplementedError(
-                'Multiple image processing is not yet supported.')
+        responses = annotator_client.batch_annotate_images(requests).responses
+        if len(responses) == 1:
```

Comment:
Please either return a list (or other iterable) all the time or return the object all the time. Do not magically flatten if there is one and exactly one result, as it is entirely common for the developer to not know how many results to expect back.

(If the Vision API returns results with an excessive amount of nesting, we can abridge that for developers by adding appropriate helper methods on `Annotations` objects.)